### PR TITLE
Feat/candidate edit mode form

### PR DESCRIPTION
**Pull request by oleksiuk-ol**
image

https://trello.com/c/FhbaQa2w/163-fe-fefactor-edit-mode-to-form

Known issues:
-Incorrect time (one hour back) is saved -FIXED
-Sometimes default time missing in the form -FIXED
-Each child in a list should have a unique "key" prop warhing -FIXED
-Hardcoded isPlanningToJoin -FIXED
-Label are empty on select -FIXED
-Some fields are missing -FIXED

### CODE REVIEW

File: src/components/candidateContent/candidateInfo/CandidateInfo.jsx [hunk 1]
```diff
@@ -9,108 +8,49 @@ import {
...
   ListItem,
   ListItemText,
-  Button,
-  Input,
 } from '@mui/material';
 import { getFieldLabel } from '../../../utils';
-import { updateCandidateInfo } from '../../../store/commands';
+
 import { tableCandidateInfoFields } from '../../../constants/tableCandidateInfoFields';
+// eslint-disable-next-line import/no-named-as-default
```
Comment (nsukhotskaya) [inline src/components/candidateContent/candidateInfo/CandidateInfo.jsx:15]: Please fix eslint problem and delete the comment

File: src/components/candidateContent/candidateInfo/CandidateInfo.jsx [hunk 2]
```diff
@@ -9,108 +8,49 @@ import {
...
-              {getFieldLabel('common.cancel')}
-            </Button>
-          </Box>
-        )}
+        <Typography paddingLeft="1%" variant="h4">
+          {`${formatedInfo.firstName} ${formatedInfo.lastName}`}
+        </Typography>
+        <CandidateInfoEdit candidateInfo={formatedInfo} />
       </Box>
+
```
Comment (nsukhotskaya) [inline src/components/candidateContent/candidateInfo/CandidateInfo.jsx:46]: Please run format

File: src/components/candidateContent/candidateInfoEdit/CandidateInfoEdit.jsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+import {
+  fetchCandidateStatusTypes,
+  fetchEnglishLevels,
+  updateCandidateInfo,
+} from '../../../store/commands';
+
+const utc = require('dayjs/plugin/utc');
+
+const checkDataReceived = (...arrays) =>
+  arrays.every((array) => array.length !== 0);
```
Comment (nsukhotskaya) [inline src/components/candidateContent/candidateInfoEdit/CandidateInfoEdit.jsx:30]:
```suggestion
  arrays.every((array) => !array.length);
```

File: src/components/candidateContent/candidateInfoEdit/CandidateInfoEdit.jsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+  const initInfo = { ...candidateInfo };
+
+  const handleClickOpen = () => {
+    initInfo.registrationDate = dayjs(
+      initInfo.registrationDate,
+      'DD/MM/YYYY HH:mm',
+    );
+    initInfo.bestContactTime = dayjs(initInfo.bestContactTime, 'HH:mm');
+    console.log(initInfo);
```
Comment (nsukhotskaya) [inline src/components/candidateContent/candidateInfoEdit/CandidateInfoEdit.jsx:77]: Remove console log statement

File: src/components/candidateContent/candidateInfoEdit/CandidateInfoEdit.jsx [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <form>
+        <Box>
+          <Button variant="outlined" onClick={handleClickOpen}>
+            {getFieldLabel('common.edit')}
+          </Button>
+          <Drawer anchor="left" open={open}>
+            <form onSubmit={formik.handleSubmit}>
+              <Box width="50vw">
```
Comment (nsukhotskaya) [inline src/components/candidateContent/candidateInfoEdit/CandidateInfoEdit.jsx:160]: Think about `%`

**Review by nsukhotskaya — APPROVED** [review]

**Review by beliavskaja — APPROVED** [review]

**Review by nsukhotskaya — APPROVED** [review]

**Pull request merged**